### PR TITLE
doc: release notes V3.1.0: Add ARM64 and IPC notes

### LINKED ISSUE #46210: Release notes for v3.1.0: arm64
As maintainer, please send a pull request filling in the "arm64" section(s) in doc/releases/release-notes-3.1.rst as soon as possible.

### PR DESCRIPTION
**Pull request by carlocaione**
Added notes for ARM64 and IPC.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

Fixes: #46210

### CODE REVIEW

File: doc/releases/release-notes-3.1.rst [hunk 1]
```diff
@@ -225,6 +225,14 @@ Architectures
 
   * AARCH64
 
+    * Add support for GICv3 for the ARMv8 Xen Virtual Machine
+    * Fix coherence issue of SMP boot code
```
Comment (mbolivar-nordic) [inline doc/releases/release-notes-3.1.rst:229]: I'm having a bit of trouble parsing this one; what does it mean?

File: doc/releases/release-notes-3.1.rst [hunk 2]
```diff
@@ -225,6 +225,14 @@ Architectures
...
+    * Add more more memory mapping types for device memory
```
Comment (mbolivar-nordic) [inline doc/releases/release-notes-3.1.rst:230]: s/more more/more/

**Comment by mbolivar-nordic** [comment]
Looks like compliance found another typo as well

**Review by mbolivar-nordic — APPROVED** [review]
Thanks!

**Review by stephanosio — APPROVED** [review]

**Pull request merged**